src/txn/txn.c

```diff
@@ -131,10 +131,11 @@ __wt_txn_release_snapshot(WT_SESSION_IMPL *session)
     txn_shared->metadata_pinned = txn_shared->pinned_id = WT_TXN_NONE;
     F_CLR(txn, WT_TXN_HAS_SNAPSHOT);
 
-    /* Clear a checkpoint's pinned ID. */
+    /* Clear a checkpoint's pinned ID and timestamps. */
     if (WT_SESSION_IS_CHECKPOINT(session)) {
         txn_global->checkpoint_txn_shared.pinned_id = WT_TXN_NONE;
-        txn_global->checkpoint_timestamp = 0;
+        txn_global->checkpoint_oldest_timestamp = WT_TS_NONE;
+        txn_global->checkpoint_timestamp = WT_TS_NONE;
```

Comment:
I'm not sure do we need to ensure checkpoint_oldest_timestamp is cleared before checkpoint_timestamp using a write barrier?

Reply:
I think it is fine to not ensure the sequence here. If checkpoint_timestamp is cleared first and we have a race, we won't pin checkpoint_oldest_timestamp as we read checkpoint_timestamp as 0. If we clear checkpoint_oldest_timestamp first and we have a race, we read non-zero checkpoint_timestamp and pin zero timestamp (checkpoint_oldest_timestamp), which is conservative but still correct.